### ISSUE TITLE
 "acme.client:Ignoring wrong Content-Type ('application/json; charset=utf-8') for JSON decodable response"

### ISSUE THREAD

**Issue by jernst** [certbot/certbot#7339]
Originally reported against pebble: https://github.com/letsencrypt/pebble/issues/271 but according to their analysis (in the comments, excerpted here), it is a certbot issue, so I'm refiling this here.

Issue:

I'm running certbot 0.32.0 against pebble 2.2.2 on localhost. `/var/log/letsencrypt/letsencrypt.log` contains the following message:
```
2019-08-19 17:56:38,338:DEBUG:acme.client:Ignoring wrong Content-Type ('application/json; charset=utf-8') for JSON decodable response
```
~~I would think that's probably certbot complaining about pebble, which is why I file this here (not sure that is correct, however).~~

This appears to be part of the register flow as it follows `https://0.0.0.0:14000 "POST /sign-me-up HTTP/1.1"`. There similar messages further below for other interactions.

Analysis:

...It looks like [this is the location](https://github.com/certbot/certbot/blob/d39f63feca9827aa9267d14869dcb72b85e40c6f/acme/acme/client.py#L1071-L1074) that produced that log line. ... comparing the full raw content type value to the string cls.JSON_CONTENT_TYPE which is defined here as the string: application/json

Pebble is sending a content type value of "application/json; charset=utf-8" which fails the string comparison because of the inclusion of the charset parameter.

**Comment by schoen** [comment]
A user on the forum mentioned another content-type issue (which possibly should be tracked separately, as it looks like it might be in the other direction with Boulder not understanding a content-type from Certbot rather than with Certbot not understanding a content-type from Pebble):

```
2019-08-14 10:25:18,749:DEBUG:urllib3.connectionpool:https://acme-v02.api.letsencrypt.org:443 “POST /acme/cert/redacted HTTP/1.1” 415 168
2019-08-14 10:25:18,750:DEBUG:acme.client:Received response:
HTTP 415
Server: nginx
Content-Type: application/problem+json
Content-Length: 168
Link: https://acme-v02.api.letsencrypt.org/directory;rel=“index”
Replay-Nonce: oksQitzt64My7oDIQhReG_G1DWKWUdJ901Yg89A9iMw
Expires: Wed, 14 Aug 2019 15:25:18 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Wed, 14 Aug 2019 15:25:18 GMT
Connection: close

{
“type”: “urn:ietf:params:acme:error:malformed”,
“detail”: "Invalid Content-Type header on POST. Content-Type must be “application/jose+json”",
“status”: 415
}
2019-08-14 10:25:18,755:DEBUG:acme.client:Error during a POST-as-GET request, your ACME CA may not support it:
urn:ietf:params:acme:error:malformed :: The request message was malformed :: Invalid Content-Type header on POST. Content-Type must be “application/jose+json”
```

**Comment by alexzorin** [comment]
Sorry for the force pushes :( .

**Comment by adferrand** [comment]
Closed by #7342
